src/wp-includes/widgets/class-wp-widget-archives.php

```diff
@@ -109,15 +109,31 @@ public function widget( $args, $instance ) {
 
 			<?php ob_start(); ?>
 <script>
-(function() {
-	var dropdown = document.getElementById( "<?php echo esc_js( $dropdown_id ); ?>" );
-	function onSelectChange() {
-		if ( dropdown.options[ dropdown.selectedIndex ].value !== '' ) {
-			document.location.href = this.options[ this.selectedIndex ].value;
-		}
+(function( dropdownId ) {
```

Comment:
Might as well use an arrow function:

```suggestion
(( dropdownId ) => {
```